Rx/v2/src/rxcpp/operators/rx-any.hpp

```diff
@@ -78,37 +113,99 @@ struct any
     }
 };
 
-template <class Predicate>
-class any_factory
-{
-    typedef rxu::decay_t<Predicate> test_type;
+}
 
-    test_type test;
-public:
-    any_factory(test_type t) : test(t) { }
+/*! @copydoc rx-any.hpp
+*/
+template<class... AN>
+auto any(AN&&... an)
+    ->     operator_factory<any_tag, AN...> {
+    return operator_factory<any_tag, AN...>(std::make_tuple(std::forward<AN>(an)...));
+}
 
-    template<class Observable>
-    auto operator()(Observable&& source)
-        -> decltype(source.template lift<rxu::value_type_t<rxu::decay_t<Observable>>>(any<rxu::value_type_t<rxu::decay_t<Observable>>, Predicate>(test))) {
-        return      source.template lift<rxu::value_type_t<rxu::decay_t<Observable>>>(any<rxu::value_type_t<rxu::decay_t<Observable>>, Predicate>(test));
-    }
-};
+/*! \brief Returns an Observable that emits true if any item emitted by the source Observable satisfies a specified condition, otherwise false. Emits false if the source Observable terminates without emitting any item.
 
-}
+    \tparam Predicate the type of the test function.
 
-template <class Predicate>
-inline auto exists(Predicate test)
-->      detail::any_factory<Predicate> {
-    return  detail::any_factory<Predicate>(test);
+    \param p the test function to test items emitted by the source Observable.
+
+    \return  An observable that emits true if any item emitted by the source observable satisfies a specified condition, otherwise false.
+
+    \sample
+    \snippet exists.cpp exists sample
+    \snippet output.txt exists sample
+*/
+template<class... AN>
+auto exists(AN&&... an)
+    ->     operator_factory<exists_tag, AN...> {
+    return operator_factory<exists_tag, AN...>(std::make_tuple(std::forward<AN>(an)...));
 }
 
-template <class T>
-inline auto contains(T value)
-->      detail::any_factory<std::function<bool(T)>> {
-    return  detail::any_factory<std::function<bool(T)>>([value](T n) { return n == value; });
+/*! \brief Returns an Observable that emits true if the source Observable emitted a specified item, otherwise false. Emits false if the source Observable terminates without emitting any item.
+
+    \tparam T the type of the item to search for.
+
+    \param value the item to search for.
+
+    \return An observable that emits true if the source Observable emitted a specified item, otherwise false.
+
+    \sample
+    \snippet contains.cpp contains sample
+    \snippet output.txt contains sample
+*/
+template<class... AN>
+auto contains(AN&&... an)
+->     operator_factory<contains_tag, AN...> {
+    return operator_factory<contains_tag, AN...>(std::make_tuple(std::forward<AN>(an)...));
 }
+
 }
 
+template<>
+struct member_overload<exists_tag>
```

Comment:
Looks Great!

Perhaps we can just change one of the `exists()` test to use `any()` to ensure that it does not get broken later.